_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,249 @@
+
+This page outlines the steps required to upload a custom model with the ML Commons plugin.
+
+## Prerequisites 
+
+To upload a custom model to OpenSearch, you'll need to prepare outside of your OpenSearch cluster. You can use a current model or train a new model depending on your needs.
+
+### Model format
+
+Although most ML models are written in Python, you'll need to export the model into a portable format. As of 2.4, OpenSearch only supports [torchscript](https://pytorch.org/docs/stable/jit.html).
+
+### Model size
+
+Most deep-learning models are over 100 MBs, making it difficult to fit the model into a single document. OpenSearch splits in portable model file into smaller chunks to store in an OpenSearch index. When allocating ML or Data nodes to your OpenSearch cluster, please be aware of the size of your model to prevent any downtime when making inferences. Furthermore, asynchronous models load faster than synchronous.
+
+## Upload model to OpenSearch
+
+Use the URL upload operation for models that already exist on another server, such as GitHub or S3. 
+
+```
+POST /_plugins/_ml/models/_upload
+```
+
+The URL upload method requires the following request fields:
+
+Field | Data Type | Description
+:---  | :--- | :--- 
+`name`| string | The name of the model. |
+`version` | integer | The version number of the model. Version number should follow standard semantic format, major.minor.patch |
+`model_format` | string | The portable format of the model file. Currently only supports `TORCH_SCRIPT`. |
+`model_config` | string | The model's configuration, including the `model_type`, `embedding_dimension`, and `framework_type`. |
+`url` | string | The URL of the server which contains the model. |
+
+#### Sample request
+
+The following sample request uploads version `1.0.0` of an NLP sentence transformation model named `all-MiniLM-L6-v2`.
+
+```json
+POST /_plugins/_ml/models/_upload
+{
+  "name": "all-MiniLM-L6-v2",
+  "version": "1.0.0",
+  "description": "test model",
+  "model_format": "TORCH_SCRIPT",
+  "model_config": {
+    "model_type": "bert",
+    "embedding_dimension": 384,
+    "framework_type": "sentence_transformers",
+  },
+  "url": "https://github.com/ylwu-amzn/ml-commons/blob/2.x_custom_m_helper/ml-algorithms/src/test/resources/org/opensearch/ml/engine/algorithms/text_embedding/all-MiniLM-L6-v2_torchscript_sentence-transformer.zip?raw=true"
+}
+```
+
+#### Sample response
+
+OpenSearch responds with the `task_id` and task `status`.
+
+```json
+{
+  "task_id" : "ew8I44MBhyWuIwnfvDIH", 
+  "status" : "CREATED"
+}
+```
+
+## Load the model
+
+The load model operation loads a model into memory by reading the model from the index. Since models indexed by OpenSearch are indexed as small chunks, the loading process can take time. 
```

Comment:
The load model API will read model chunks from ML model index then create model instance and cache it in memory. Bigger model will have more chunks, so the loading process will take more time.